scheduler/scheduler-smartproxy-common/src/main/java/org/ow2/proactive/scheduler/smartproxy/common/JobDatabase.java

```diff
@@ -190,44 +190,52 @@ public AwaitedJob getAwaitedJob(String id) {
      * if a InvalidClassException occur, we clean the database
      */
     public void loadJobs() {
+        this.loadJobs(true);
+    }
+
+    private void loadJobs(boolean firstAttempt) {
         try {
             writeLock.lock();
-            if (recMan != null) {
-                try {
-                    recMan.close();
-                } catch (Exception e) {
-                }
-            }
+            closeRecordManager();
             try {
                 recMan = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
                 awaitedJobs = recMan.hashMap(STATUS_RECORD_NAME);
                 // This empty loop triggers InvalidClassException in case of serial version uid problems
                 for (Map.Entry<String, AwaitedJob> job : awaitedJobs.entrySet())
                     ;
                 recMan.commit();
-            } catch (IOError e) {
-                // we track invalid class exceptions
-                if (e.getCause() instanceof InvalidClassException) {
-                    try {
-                        recMan.close();
-                    } catch (IOException e1) {
-
-                    }
+                if (!firstAttempt) {
+                    log.info("Loading of job database successful after clean.");
+                }
+            } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)